internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -380,7 +380,15 @@ func createWindowsContainerDocument(ctx context.Context, coi *createOptionsInter
 	if err != nil {
 		return nil, nil, err
 	}
-	v2Container.AdditionalDeviceNamespace = extensions
+	// Do not add empty extensions. It causes container creation to
+	// fail with invalid config error.
+	if len(extensions.DeviceExtension) > 0 {
```

Comment:
I think it's probably better to make `getDeviceExtensions` return `nil` when there are no device extensions, rather than ` ContainerDefinitionDevice` with an empty `DeviceExtension` field.

We could also probably do this version check inside that function as well. Then the code here doesn't have to change.

Reply:
Fixed.